pydoctor/templatewriter/__init__.py

```diff
@@ -151,26 +194,73 @@ class _StaticTemplate(Template):
 
     For CSS and JS templates.
     """
+    data: bytes
+
     @property
     def version(self) -> int:
         return -1
     @property
     def loader(self) -> None:
         return None
+    
+    def write(self, output_dir: Path, subfolder: Optional[PurePath] = None) -> PurePath:
+        """
+        Directly write the contents of this static template as is to the output dir.
+
+        @returns: The relative path of the file that has been wrote.
+        """
+        _subfolder_path = subfolder if subfolder else PurePath()
+        _template_path = _subfolder_path.joinpath(self.name)
+        outfile = output_dir.joinpath(_template_path)
+        self._write(outfile)
+        return _template_path
+    
+    def _write(self, path: Path) -> None:
+        with path.open('wb') as fobjb:
+            fobjb.write(self.data)
+
+class _TemplateSubFolder(_StaticTemplate):
+    """
+    Special template to hold a subfolder contents. 
```

Comment:
You suggest to `PurePath` instead of the name? But we would need to store folder entries as well, because we need to `mkdir` them at some point in the build process, this is why having a special object to represent a folder makes sens to me.
The implementation of `is_empty` should perhaps be done on the subclass as well.

Reply:
You could do the same thing Git does: don't track directories, instead only create a directory when you need to write a file in it.